sklearn/feature_selection/rfe.py

```diff
@@ -197,8 +219,29 @@ def _fit(self, X, y, step_score=None):
             # for sparse case ranks is matrix
             ranks = np.ravel(ranks)
 
+            # Adjust step using special parameters if specified
+            if self.step_one_threshold is not None:
```

Comment:
I still would appreciate making this less nested and having a clearer interface by pulling it out into a method or generator

Reply:
After thinking about it for a bit and trying to make it more concise i don’t think it’s possible unless I’m missed some possible way.  In order to perform the logic necessary it actually takes those if statements and nested like that. Remember that it has to handle when only reducing_step is set, or only step_one_threshold, or both parameters.

Reply:
I'm not too fussed about this. It is cosmetic. I'm just finding it harder to read than I think it should be.